### ISSUE TITLE
Integrated vector art boob scaling

### ISSUE THREAD

**Issue by freecitiesdev** [Free-Cities/Free-Cities#950]
In Boob.tw, the following line sets up the scaling:

<<set _art_scale_factor = 0.190308*Math.pow(_artSlave.boobs,(1.0/3.0))-0.205304 >>

I'd like to understand what this is doing generally, and I'd like to play around with the scaling, since huge tits are scaled a bit big right now. What are the different parts of this affecting?

**Comment by prndev** [comment]
Before continuing, it is good to know that the actual boob display scaling is done with [affine transformation](https://en.wikipedia.org/wiki/Affine_transformation). I could not find a decent video at youtube about this topic. Basically, we need to know that the transformation is defined by a matrix with six parameters. In our case, two parameters are always zero, two parameters define define the scaling and two parameters define a movement.

_art_scale_factor is a linear scaling factor. A value of 1.0 means "keep the default size as it is drawn in vector_source.svg". A value of 2.0 means "display with double size" and so on.
In the SVG-editor Inkscape, you can have this value displayed by the tool Object →Transformation → Matrix → Edit Current Matrix. The scaling appears twice (for height and width) as the matrix values A and D.  
![boob_scaling_matrix](https://user-images.githubusercontent.com/25567918/29112826-62908924-7cef-11e7-8acb-2e3dd5c574ff.png)  
I wrote down the boob volumes and the corresponding scaling factors for Nov-X original boob sizes 1 to 7 and threw them into gnuplot's function fitter for a cubic root function: *a ∛x + b*. I had the notion, a cubic root function would be good as volume is a cubic thing and a scaling factor is a linear thing. gnuplot calculated the parameters *a = 0.190308* and *b = -0.205304*. I simply used them as they came. This is how this line of code was devised.  
For the fitting, one could do the same with an online-tool like [WolframAlpha](https://www.wolframalpha.com/input/?i=log+fit+%7B%7B250,0%7D,%7B1000,1.69%7D,%7B2000,2.18%7D,%7B4000,2.8%7D,%7B10000,3.88%7D%7D). WolframAlpha does not seem to support cubic root fitting, though. Allegedly, you can even use Microsoft Excel, but please don't ask me how that works.

Unfortunately, scaling an SVG object also changes its position. That is why the linear translation factors (matrix parameters E and F) are needed, too. For a scaling factor of 1.0, they should be exactly 0. The current formulae should be okay as long as the position of boobs within vector_source.svg is not changed. They do not need to be updated if the formula for _art_scale_factor is changed.

**Long story short:** You can modify this line of code in any way you want. :)  
However, to be consistent with the original art display, for boobs of 250cc volume the _art_scale_factor should have a value of about 1.0.  
For displaying huge boobs less big, you can lower the value of *a* to let's say 0.15 and then solve *1.0 = 0.15 ∛250 + b* for *b*. Alternatively, you can use a higher root function. Or a logarithm. There are many possibilities. :)

I hope my explanations are helpful.

**Comment by freecitiesdev** [comment]
That's considerably more involved than I thought, but I did understand and expect that there was an issue with scaling SVGs changing their position.

I approve strongly of your use of a cube function, since CCs are a measure of volume. If you're willing to help recalculating this, I think the useful target here is that some of our resident pregnancy modders found that a pregnancy typically reaches ~15,000CC. Therefore, the pregnant belly vector should roughly match 15,000CC boobs. If not, that's fine; I'll use your explanation as a guide. Thanks either way.

**Comment by prndev** [comment]
Does #958 please your eye (or any other appropriate body-part)?
Display of boobs starts at 300cc per boob. A single boob of 15000cc should approximately be the same size as a belly. I fiddled around and resorted to a [logarithmic function](https://www.wolframalpha.com/input/?i=log+fit+%7B%7B300,1%7D,%7B15000,2.5%7D%7D). Visually, small boobs now grow faster and large boobs now grow slower. To me, it looks okay from cup sizes A to F. Everything above is fantasy porn anyway. ;)
